src/TestExplorer/TestParsers/SwiftTestingOutputParser.ts

```diff
@@ -0,0 +1,186 @@
+import * as readline from "readline";
+import { Readable } from "stream";
+import {
+    INamedPipeReader,
+    UnixNamedPipeReader,
+    WindowsNamedPipeReader,
+} from "./TestEventStreamReader";
+import { ITestRunState } from "./TestRunState";
+
+// All events produced by a swift-testing run will be one of these three types.
+export type SwiftTestEvent = MetadataRecord | TestRecord | EventRecord;
+
+interface VersionedRecord {
+    version: number;
+}
+
+interface MetadataRecord extends VersionedRecord {
+    kind: "metadata";
+    payload: Metadata;
+}
+
+interface TestRecord extends VersionedRecord {
+    kind: "test";
+    payload: Test;
+}
+
+export type EventRecordPayload =
+    | RunStarted
+    | TestStarted
+    | TestEnded
+    | TestCaseStarted
+    | TestCaseEnded
+    | IssueRecorded
+    | TestSkipped
+    | RunEnded;
+
+export interface EventRecord extends VersionedRecord {
+    kind: "event";
+    payload: EventRecordPayload;
+}
+
+interface Metadata {
+    [key: string]: object; // Currently unstructured content
+}
+
+interface Test {
+    kind: "suite" | "function" | "parameterizedFunction";
+    id: string;
+    name: string;
+    _testCases?: TestCase[];
+    sourceLocation: SourceLocation;
+}
+
+interface TestCase {
+    id: string;
+    displayName: string;
+}
+
+// Event types
+interface RunStarted {
+    kind: "runStarted";
+}
+
+interface RunEnded {
+    kind: "runEnded";
+}
+
+interface BaseEvent {
+    timestamp: number;
+    messages: EventMessage[];
+    testID: string;
+}
+
+interface TestStarted extends BaseEvent {
+    kind: "testStarted";
+}
+
+interface TestEnded extends BaseEvent {
+    kind: "testEnded";
+}
+
+interface TestCaseStarted extends BaseEvent {
+    kind: "testCaseStarted";
+}
+
+interface TestCaseEnded extends BaseEvent {
+    kind: "testCaseEnded";
+}
+
+interface TestSkipped extends BaseEvent {
+    kind: "testSkipped";
+}
+
+interface IssueRecorded extends BaseEvent {
+    kind: "issueRecorded";
+    sourceLocation: SourceLocation;
+}
+
+export interface EventMessage {
+    text: string;
+}
+
+export interface SourceLocation {
+    _filePath: string;
+    line: number;
+    column: number;
+}
+
+export class SwiftTestingOutputParser {
+    private completionMap = new Map<number, boolean>();
+
+    /**
+     * Watches for test events on the named pipe at the supplied path.
+     * As events are read they are parsed and recorded in the test run state.
+     */
+    public async watch(
+        path: string,
+        runState: ITestRunState,
+        pipeReader?: INamedPipeReader
```

Comment:
Is this just for testing?

Reply:
Yep, see https://github.com/plemarquand/vscode-swift/blob/0214e8dd8e7f314b1f54fbb01ee5568e05fbcada/test/suite/testexplorer/SwiftTestingOutputParser.test.ts#L27 for the test implementation